jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketConnection.java

```diff
@@ -331,26 +331,20 @@ public void failed(Throwable cause)
 
     private void acquireNetworkBuffer()
     {
-        try (AutoLock l = lock.lock())
-        {
-            if (networkBuffer == null)
-                networkBuffer = newNetworkBuffer(getInputBufferSize());
-        }
+        if (networkBuffer == null)
+            networkBuffer = newNetworkBuffer(getInputBufferSize());
     }
 
     private void reacquireNetworkBuffer()
     {
-        try (AutoLock l = lock.lock())
-        {
-            if (networkBuffer == null)
-                throw new IllegalStateException();
+        if (networkBuffer == null)
+            throw new IllegalStateException();
 
-            if (networkBuffer.getByteBuffer().hasRemaining())
-                throw new IllegalStateException();
+        if (networkBuffer.getByteBuffer().hasRemaining())
+            throw new IllegalStateException();
 
-            networkBuffer.release();
-            networkBuffer = newNetworkBuffer(getInputBufferSize());
-        }
+        networkBuffer.release();
+        networkBuffer = newNetworkBuffer(getInputBufferSize());
```

Comment:
@lachlan-roberts not for this PR, but we should do the same as HTTP/1 and reuse the existing buffer, even if retained, if there is sufficient space in it to do something useful. See `org.eclipse.jetty.server.internal.HttpConnection#parseAndFillForContent`